src/components/Chat/ChatDetails.tsx

```diff
@@ -0,0 +1,179 @@
+        //this.state = this.blankState();
+        //let player = player_cache.lookup(this.props.playerId);
+        let channel = this.props.chatChannelId;
+        if (channel) {
+            this.state = {
+                channelId: channel,
+            };
+        }
+
+        ////if (player) {
+        //    this.state = Object.assign(this.state, player);
+        //}
+    }
+
+    // UNSAFE_componentWillMount()  {
+        //this.resolve(this.props.playerId);
+        // this.resolve(this.props.chatChannel);
+    // }
+
+    //blankState() {
+        //return {
+        //    resolved: false,
+        //    resolving: 0,
+        //    username: "...",
+        //    //icon: data.get('config.cdn_release') + '/img/default-user.svg',
+        //    icon: "",
+        //    ranking: "...",
+        //    rating: "...",
+        //    ui_class: "...",
+        //    country: "un",
+        //    error: null,
+        //};
+    //}
+    //resolve(player_id) {
+        //this.setState({resolved: false});
+        //player_cache.fetch(
+        //    this.props.playerId,
+        //    [
+        //        "username",
+        //        "icon",
+        //        "ratings",
+        //        "pro",
+        //        "country",
+        //        "ui_class",
+        //    ]
+        //)
+        //.then((player) => {
+        //    this.setState(Object.assign({resolved: true}, player as any));
+        //})
+        //.catch((err) => {
+        //    if (player_id === this.props.playerId) {
+        //        this.setState({resolved: true, error: _("Error loading player information")});
+        //        console.error(err);
+        //    }
+        //});
+    //}
+    //UNSAFE_componentWillReceiveProps(new_props) {
+        //if (new_props.playerId !== this.props.playerId) {
+        //    let player = player_cache.lookup(new_props.playerId);
+        //    let new_state = this.blankState();
+        //    if (player) {
+        //        new_state = Object.assign(new_state, this.state, player);
+        //    }
+        //    this.setState(new_state);
+        //    setTimeout(() => {
+        //        this.resolve(new_props.playerId);
+        //    }, 1);
+        //}
+    //}
+    //componentWillUnmount() {
+    //}
+
+    close_all_modals_and_popovers = () => {
+        close_all_popovers();
+        close_friend_list();
+    }
+
+    leave = (_ev) => {
+        //Chat.part(this.channel, false, false);
+        //leaveActiveChannel(); //figure out how to ask chat to leave channel
+        this.close_all_modals_and_popovers();
+    }
+    goToGroup = (_ev) => {
+        browserHistory.push('/group/' + this.state.channelId);
+        this.close_all_modals_and_popovers();
+    }
+    goToTournament = (_ev) => {
+        browserHistory.push('/tournament/' + this.state.channelId);
+        this.close_all_modals_and_popovers();
+    }
+
+    render() {
+        let user = data.get("user");
+
+        console.log("I'm rendering");
```

Comment:
this line gets printed but none of the divs show up on the screen

Reply:
@anoek  I'm trying to use the player name menu as a template to build the chat options menu but it's not rendering even though the render() code is being run... am I missing something obvious?

Reply:
They are rendered at the bottom left of the page. If you click the ▼, the site suddenly has a scroll bar.

Reply:
> am I missing something obvious?


> bottom left of the page. ...  the site suddenly has a scroll bar.

Well, that certainly qualifies!!! 🤣 





> You specified the below property in `popover` as below chat window.

Hmm, odd. I copied that from display_details in Player.tsx and they load directly under the username. (Hence, I figured, "popover" rather than "popinline")